.github/workflows/ci.yml

```diff
@@ -10,16 +10,31 @@ on:
       - main
       - '*.x'
 
+env:
+  circt: "firtool-1.48.0"
+
 jobs:
+  set-circt-versions:
+    name: Set the CIRCT version from the env
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Set CIRCT Version
+        id: set-circt-version
+        run: |
+          echo versions='["${{ env.circt }}"]' >> $GITHUB_OUTPUT
+    outputs:
+      circt-versions: ${{ steps.set-circt-version.outputs.versions }}
+
   ci:
     name: ci
+    needs: [set-circt-versions]
     strategy:
       matrix:
         system: ["ubuntu-20.04"]
         jvm: [8]
         scala: ["2.13.11"]
         espresso: ["2.4"]
-        circt: ["firtool-1.48.0"]
+        circt: ${{ fromJSON(needs.set-circt-versions.outputs.circt-versions) }}
```

Comment:
All of this is necessary because `env` is not accessible when inside the `matrix` context. The `fromJSON` approach lets the CIRCT version still be specified in one place---in the `env`---and used in both the matrix construction and in the publish job (which does have access to the `env` context).